velox/common/caching/CachedFactory.h

```diff
@@ -54,8 +54,11 @@ class CachedFactory {
   // true, but performance will suffer.
   CachedFactory(
       std::unique_ptr<SimpleLRUCache<Key, Value>> cache,
-      std::unique_ptr<Generator> generator)
-      : cache_(std::move(cache)), generator_(std::move(generator)) {}
+      std::unique_ptr<Generator> generator,
+      const bool cacheEnabled)
```

Comment:
Do we need this boolean? Can we instead allow 'cache' to be null and use this as an indication that cache is disabled?

We need to update comments as well.